lib/msf/core/auxiliary/mikrotik.rb

```diff
@@ -0,0 +1,340 @@
+            cred[:address] = value['address']
+            cred[:protocol] = 'tcp'
+            cred[:username] = value['user']
+            cred[:private_data] = value['password']
+            create_credential_and_login(cred)
+          end
+          #
+          # Wireless networks details
+          #
+        when '/interface wireless security-profiles'
+          # https://wiki.mikrotik.com/wiki/Manual:Interface/Wireless#Security_Profiles
+          # add name=openwifi supplicant-identity=MikroTik
+          # add authentication-types=wpa-psk mode=dynamic-keys name=wpawifi supplicant-identity=MikroTik wpa-pre-shared-key=presharedkey
+          # add authentication-types=wpa2-psk mode=dynamic-keys name=wpa2wifi supplicant-identity=MikroTik wpa2-pre-shared-key=presharedkey
+          # add authentication-types=wpa2-eap mode=dynamic-keys mschapv2-password=password mschapv2-username=username name=wpaeapwifi supplicant-identity=MikroTik
+          # add mode=static-keys-required name=wepwifi static-key-0=0123456789 static-key-1=0987654321 static-key-2=1234509876 static-key-3=0192837645 supplicant-identity=MikroTik
+          values.each do |value|
+            next unless value.starts_with?('add ')
+            value = values_to_hash(value)
+            output = "#{thost}:#{tport} Wireless AP #{value['name']}"
+
+            cred = credential_data.dup
+            if !value['authentication-types'] && (!value['mode'] || value['mode'] == 'none') # open wifi
+              next
+            elsif value['wpa-pre-shared-key'] && !value['wpa-pre-shared-key'].empty?
+              output << " with WPA password #{value['wpa-pre-shared-key']}"
+              cred[:private_data] = value['wpa-pre-shared-key']
+            elsif value['wpa2-pre-shared-key'] && !value['wpa2-pre-shared-key'].empty?
+              output << " with WPA2 password #{value['wpa2-pre-shared-key']}"
+              cred[:private_data] = value['wpa2-pre-shared-key']
+            elsif value['authentication-types'] == 'wpa2-eap'
+              output << " with WPA2-EAP username #{value['mschapv2-username']} password #{value['mschapv2-password']}"
+              cred[:username] = value['mschapv2-username']
+              cred[:private_data] = value['mschapv2-password']
+            elsif value['static-key-0'] || value['static-key-1'] || value['static-key-2'] || value['static-key-3']
+              (0..3).each do |i|
+                key = "static-key-#{i}"
+                next unless value[key]
+                output << " with WEP password #{value[key]}"
+                cred[:private_data] = value[key]
+                create_credential_and_login(cred) # run for each key we find
+              end
+            end
+
+            print_good(output)
+            unless value['static-key-0'] || value['static-key-1'] || value['static-key-2'] || value['static-key-3']
+              create_credential_and_login(cred)
+            end
+          end
```

Comment:
I debated that, but thought that since it didn't have a credential, it wouldn't be worth printing.  In theory a lot of things could be printed from a config, I just chose to do things with creds or version numbers.

No real coin in the game though, one line print statement is easy to change if anyone else thinks it would be a good thing to print.

Reply:
decided on a `vprint_good`

Reply:
also added a spec test for openwifi